src/wfs/src/test/java/org/geoserver/wfs/GetFeatureTest.java

```diff
@@ -375,7 +377,25 @@ public void testRequestDisabledResource() throws Exception {
                 doc);
         XMLAssert.assertXpathEvaluatesTo("typeName", "//ogc:ServiceException/@locator", doc);
     }
-    
+
+    /**
+     * Test that a request for a resource from a disabled store fails.
+     */
+    @Test
+    public void testRequestDisabledStore() throws Exception {
+        Catalog catalog = getCatalog();
+        StoreInfo store = catalog.getStoreByName("cdf", DataStoreInfo.class);
+        store.setEnabled(false);
```

Comment:
@aaime I have force-pushed an updated version that re-enables the store in a ```finally``` block to restore the fixture state. Our tests should not rely on luck!